src/891b0_len_fb0.c

```diff
@@ -149,13 +148,59 @@ void func_800F0864(void) {
         D_801094A8 = !D_801094A8;
     }
 }
-#else
-INCLUDE_ASM(s32, "891b0_len_fb0", func_800F0864);
-#endif
 
-INCLUDE_ASM(s32, "891b0_len_fb0", func_800F09EC);
+void func_800F09EC(void) {
+    PlayerStatus* playerStatus = &gPlayerStatus;
+    f32 sin, cos;
+
+    if (playerStatus->actionState != ACTION_STATE_WALK &&
+        playerStatus->actionState != ACTION_STATE_RUN &&
+        (playerStatus->actionState != ACTION_STATE_SPIN || playerStatus->actionSubstate != 0) &&
+        ((playerStatus->actionState != ACTION_STATE_LAND && playerStatus->actionState != ACTION_STATE_IDLE) ||
+          playerStatus->flags >= 0)
+    ) {
+        D_801094AC = 4;
+        return;
+    }
 
-INCLUDE_ASM(s32, "891b0_len_fb0", func_800F0B3C);
+    if (D_801094AC++ >= 4) {
+        D_801094AC = 0;
+        sin_cos_rad((clamp_angle(-playerStatus->currentYaw) * TAU) / 360.0f, &sin, &cos);
```

Comment:
Should use `DEG_TO_RAD` introduced in #773 for these 
```suggestion
        sin_cos_rad(clamp_angle(DEG_TO_RAD(-playerStatus->currentYaw), &sin, &cos);
```